_posts/2019-04-21-dual-boot-with-rails-6-0-beta.md

```diff
@@ -0,0 +1,197 @@
+1 gem installed
+```
+
+_Warning: `ten_years_rails` requires Ruby 2.3 or higher. You can find a manual
+workaround below._
+
+Now that I can use that gem, I will init my `Gemfile.next` file like this:
+
+```
+$ next --init
+Created Gemfile.next (a symlink to your Gemfile). Your Gemfile has been modified
+to support dual-booting!
+
+There's just one more step: modify your Gemfile to use a newer version of Rails
+using the `next?` helper method.
+
+For example, here's how to go from 5.2.3 to 6.0:
+
+if next?
+  gem "rails", "6.0.0"
+else
+  gem "rails", "5.2.3"
+end
+```
+
+That command creates a `Gemfile.next` symlink to my `Gemfile` and adds a util
+method to my `Gemfile`:
+
+```ruby
+# Gemfile
+def next?
+  File.basename(__FILE__) == "Gemfile.next"
+end
+
+source 'https://rubygems.org'
+# ...
+```
+
+If you have any problems installing `ten_years_rails`, you can manually add the
+`next?` method to your `Gemfile` and create a symlink like this:
+
+```
+$ cd path/to/project
+$ ln -s Gemfile Gemfile.next
+```
+
+## Bump Rails (Gemfile.next)
+
+In this simple example, I only need to upgrade `rails` (from
+[Rails 4.2 to Rails 5.0](https://www.ombulabs.com/blog/rails/upgrades/upgrade-rails-from-4-2-to-5-0.html)).
+It's _very likely_ that you will have to upgrade more dependencies. My `Gemfile`
+now looks like this:
+
+```ruby
+def next?
+  File.basename(__FILE__) == "Gemfile.next"
+end
+
+source 'https://rubygems.org'
+
+# Load environment variables
+gem 'dotenv-rails', :require => 'dotenv/rails-now'
+
+if next?
+  gem 'rails', '~> 6.0.0'
+else
+  gem 'rails', '~> 5.2.3'
+end
+
+gem 'pg', '< 0.21'
+gem 'authlogic'
+# ...
+```
+
+Now I can install my current dependencies with `bundle install` and my _future_
+dependencies with `next bundle install`. If `next bundle install` doesn't work
+for you, you can just run `BUNDLE_GEMFILE=Gemfile.next bundle install`.
+
+## Run Tests
+
+After running `next bundle install`, I have a brand new `Gemfile.next.lock` file.
+That means that my dependencies are ready to run my test. So I can run them like
+this:
+
+```
+$ next bundle exec rake
+```
+
+The main advantage of using _dual booting_ your Rails application is that you
+can run the tests with two different versions of Rails. Running `bundle exec rake`
+still works thanks to the conditionals in your `Gemfile`.
+
+## Setup Continuous Integration
+
+Depending on the type of project, we like to use [Travis CI](https://travis-ci.com)
+or [Circle CI](https://circleci.com). So below you will find a couple of sample
+configuration files that you could use.
+
+Both samples will require you to commit and push `Gemfile.next` to your repository;
+will run a build matrix; and might need some tweaking.
+
+## Circle CI
+
+For all of our client projects we prefer this continuous integration service.
+Here is the configuration that you could use for dual booting in Circle CI:
+
+<script src="https://gist.github.com/etagwerker/14c9045788d356cbb797dcb5f678b135.js"></script>
```

Comment:
@etagwerker You should wrap this in a snippet, or show only the link to the configuration. Otherwise the link is broken when you click on it.

Reply:
@lubc What do you mean? The snippets show in my local environment. Could you add a screenshot of the error you're seeing? 

I'm seeing this with both snippets:

<img width="1249" alt="Screen Shot 2019-04-24 at 10 33 58 PM" src="https://user-images.githubusercontent.com/17584/56705960-28d17080-66e1-11e9-85d8-0e57f0d66beb.png">
<img width="1329" alt="Screen Shot 2019-04-24 at 10 33 44 PM" src="https://user-images.githubusercontent.com/17584/56705961-28d17080-66e1-11e9-9f7b-8df0ed12f81d.png">

Reply:
Ah, got it 👍 . I reviewed it using the built in markdown preview on github and it was a bit broken there.